wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/bootstrap/AbstractBootstrapper.java

```diff
@@ -86,6 +86,9 @@ public void bootstrap() throws InfrastructureException {
         if (finishEventFuture != null) {
             throw new IllegalStateException("Bootstrap method must be called only once.");
         }
+        if (!hasInstallersToBootstrap()) {
```

Comment:
Bootstrapper works with each machine, not the whole runtime, InternalRuntime knows nothing about bootstrapper, the bootstrapper is not part of SPI contract. So I do think it is better to not introduce bootstrapper in InternalRuntime